proposed/activity-improvements.md

```diff
@@ -0,0 +1,290 @@
+        public override int GetHashCode()
+    }
+```
+
+ ### ActivityLink
+```C#
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// Activity may be linked to zero or more other <see cref="ActivityContext"/> that are causally related.
+    /// Links can point to ActivityContexts inside a single Trace or across different Traces.
+    /// Links can be used to represent batched operations where a Activity was initiated by multiple initiating Activities,
+    /// each representing a single incoming item being processed in the batch.
+    public readonly struct ActivityLink
+    {
+        public ActivityLink(ActivityContext context) 
+        public ActivityLink(ActivityContext context, IDictionary<string, object>? attributes)
+        public ActivityContext Context { get; }
+        public IDictionary<string, object>? Attributes { get; }
+    }
+}
+```
+
+### Activity
+
+```C#
+namespace System.Diagnostics
+{
+    public partial class Activity : IDisposable // added IDisposable 
+    {
+        ....
+        public System.Diagnostics.ActivityKind Kind { get; set; }
+        public System.Collections.Generic.IEnumerable<ActivityLink> Links { get; }
+        public System.Diagnostics.Activity AddLink(ActivityLink link) 
+
+        public static System.Diagnostics.Activity? StartNew(string name) 
+        public static System.Diagnostics.Activity? StartNew(string name, ActivityContext context, IEnumerable<ActivityLink>? links = null, DateTimeOffset statTime = default) 
+        public static System.Diagnostics.Activity? StartNew(string name, ActivityContext context, ActivityKind kind, IEnumerable<System.Collections.Generic.KeyValuePair<string,string>>? tags = null, IEnumerable<ActivityLink>? links = null, DateTimeOffset startTime = default) 
+
+        public bool TryGetContext(out ActivityContext context)
+
+        public static IDisposable StartListening(
+                                                 Func<string, ActivityContext, ActivityKind, IEnumerable<System.Collections.Generic.KeyValuePair<string,string>>?, IEnumerable<ActivityLink>?, bool> shouldCreateActivity, 
+                                                 Action<Activity> onActivityStarted, 
+                                                 Action<Activity> onActivityStopped);
+
+        public void SetCustomProperty(string propertyName, object? propertyValue)
+        public object? GetCustomProperty(string propertyName) 
+
+        public void Dispose() 
+    }
+}
+
+```
+
+Q & A 
+================
+
+[TODO] There is a bunch of discussion in https://github.com/dotnet/runtime/issues/31373 that hasn't been replicated here. If those discussions continue it is probably worthwhile to put conclusions (with needed context here)
+
+
+
+Q\) It would be great if we put some guidance here regarding the Activity naming convention
+
+[Noah] Agreed. It wasn't an area I focused on much in that experiment, but we will need to have a plan for it.
+
+
+
+
+
```

Comment:
Q&A issue - performance - continuing a discussion in the [old thread](https://github.com/dotnet/runtime/issues/31373#issuecomment-600397471). Hopefully this is fair summary of our discussion there:

Previous iterations of the API design included an ActivitySource object that would cache a set of listeners interested in a particular activity name. This meant that emitting an Activity only has cost proportional to the number of interested listeners and no string name comparison needs to occur. The current design does not have this optimization so emitting an Activity has cost proportional to the total number of listeners in the process and string comparisons are needed each time. @pakrym and @benaadams were concerned the performance will be inadequate and we need to bring back some optimization. My position is that the optimization adds complexity and cost for limited benefit given the scale and performance needs of the scenarios I can envision. I hope I am not being overbearing here. If there is a scenario where this matters I want to do it but I also want to stay within the realm of what we believe people are likely to do so that we are spending our budget of time and complexity judiciously. @pakrym and @benaadams it would still be great to get your thoughts - at this point I am unsure if you find my rationale makes sense or you are still worried or something else? Thanks!